File: phoenix_app/priv/repo/migrations/20241101000001_create_todos.exs
defmodule TodoBoard.Repo.Migrations.CreateTodos do
  use Ecto.Migration

  def change do
    create table(:todos, primary_key: false) do
      add :id, :binary_id, primary_key: true
      add :room_id, :string, null: false
      add :content, :string, null: false
      add :completed, :boolean, default: false, null: false
      add :position, :integer, default: 0, null: false

      timestamps()
    end

    create index(:todos, [:room_id])
    create index(:todos, [:room_id, :inserted_at])
  end
end

File: rails_app/db/migrate/20241101000001_create_todos.rb
class CreateTodos < ActiveRecord::Migration[8.0]
  def change
    create_table :todos, id: :uuid do |t|
      t.string :room_id, null: false
      t.string :content, null: false
      t.boolean :completed, default: false, null: false
      t.integer :position, default: 0, null: false

      t.timestamps
    end

    add_index :todos, :room_id
    add_index :todos, [:room_id, :created_at]
  end
end
